pkg/asset-manager-utils/test/RewardsAssetManager.test.ts

```diff
@@ -94,10 +95,29 @@ describe('Rewards Asset manager', function () {
   });
 
   describe('setConfig', () => {
-    let poolController: SignerWithAddress;
+    let assetManager: Contract;
+    let pool: SignerWithAddress;
+    let poolId: string;
 
     sharedBeforeEach(async () => {
-      poolController = lp; // TODO
+      pool = lp;
+
+      const tx = await vault.connect(pool).registerPool(GeneralPool);
+      const receipt = await tx.wait();
+      const event = expectEvent.inReceipt(receipt, 'PoolRegistered');
+      poolId = event.args.poolId;
+
+      // Deploy Asset manager
+      assetManager = await deploy('MockRewardsAssetManager', {
+        args: [vault.address, ZERO_BYTES32, tokens.DAI.address],
+      });
+
+      // Assign assetManager to the DAI token
+      const assetManagers = [assetManager.address, ZERO_ADDRESS];
+
+      await vault.connect(pool).registerTokens(poolId, tokens.addresses, assetManagers);
+
+      await assetManager.initialise(poolId);
```

Comment:
Why are we setting up a Pool again here from an EOA? Can't the one created in the `setup()` function be used?

Reply:
Purely because I did this and then realised I needed to create a MockPool for the rebalancing tests and haven't gotten round to updating this again. Will carve this out now,